libwild/src/args.rs

```diff
@@ -711,6 +745,25 @@ pub(crate) fn parse<F: Fn() -> I, S: AsRef<str>, I: Iterator<Item = S>>(input: F
         args.allow_copy_relocations = false;
     }
 
+    let should_set_no_undefined = match args.output_kind() {
```

Comment:
There are other ways this could be written that might be simpler. Up to you if you think it's easier to read.

```rust
args.no_undefined |= matches!((args.output_kind(), args.unresolved_symbols),
  (OutputKind::SharedObject, UnresolvedSymbols::IgnoreInObjectFiles) |
  (_, UnresolvedSymbols::IgnoreInSharedLibs | UnresolvedSymbols::ReportAll)
);
```